### ISSUE TITLE
Unable to select "My key" from openkeychain

### ISSUE THREAD

**Issue by vikasrawal** [thunderbird/thunderbird-android#2601]
I have installed the latest version of K9 mail and OpenKeychain from google play store. I have imported my private key and a few public keys in OpenKeychain.  In K9 mail account settings, in Cryptography,  I chose OpenKeychain as the OpenPGP app. But it still does not allow me to select anything under "My Key". Nothing happens when I press that option.

**Comment by Taijian** [comment]
Similar problem here. For me, tapping on one of the keys I have imported into OpenKeychain simply crashes K9.

K9 version: 5.207 from F-Droid
Android: 6.0.1 (patch level 2017-05-01)
Fairphone 2

**Comment by vikasrawal** [comment]
Any hope that this would be attended to anytime soon?

**Comment by Sebafox** [comment]
I got exactly the same issue as vikasrawal.

K9-Version: 5.208
Android: 7.1.1 (patch level 2017-06-01)
Lenovo Yoga Book

**Comment by anton-tsyganenko** [comment]
My friend has the same problem.

Android 5.1 (patch 2017-10-05), Flyme 6.2.0.0G
Meizu m3 note
K-9 5.403 from google play
OpenKeychain 4.9.1 (49100)

**Comment by jeaye** [comment]
Yep, I'm seeing this with a fresh K-9 and OpenKeychain from Google Play on a new Pixel 2 XL (Android 8.1.0). I see my key listed when I choose to select my account's key, but, when I tap it, the window is dismissed yet nothing is selected.

**Comment by bodtx** [comment]
Same thing here on last version of K9 and openKeychain, when I click to select my key the modal of openkeychain blinks then it desapears quickly and then nothing.

**Comment by LorenzoLMP** [comment]
Same behaviour with:
Phone: Meizu MX5 (patch 5 April 2017)
OS: Android 5.1
OpenKeychain: 5.1.4
K-9 Mail: 5.403

**Comment by momoblydblk** [comment]
Same problem here:
Chose OpenKeyChain from general settings, but I cant choose my key from my account. It keeps saying that I didnt select a key. When I pressed on it, nothing happened. Saw K9 in OpenKeyChain app.

Phone: Xiaomi Mi9
OS: MIUI 10 | Android 9
System Language: zh-CN
OpenKeyChain: 5.3 (53304)
K9: 5.600

Thanks

**Comment by PhyberApex** [comment]
I have given the persmission to open pop-ups from background but still no luck. I am on MIUI 11 tho.

~Cheers

**Comment by madman972** [comment]
I'm also with MIUI 11, and I could get the OpenKeyChain prompt after giving it the proper permissions "Display pop-up windows while running in the background".

**Comment by mkirisame** [comment]
Yes this is still a big issue, I have attached a screenshot on how to fix it manually according to previous posters:

![Screenshot_2020-02-10-19-19-32-201_com miui securitycenter](https://user-images.githubusercontent.com/37766263/74177674-874ce880-4c3a-11ea-9a14-29f0c8c82aa3.jpg)
![Screenshot_2020-02-10-19-19-37-583_com miui securitycenter](https://user-images.githubusercontent.com/37766263/74177691-8d42c980-4c3a-11ea-88f0-5e1553553499.jpg)

Sorry for non-english version, but visual people will find it anyway based on screen offset.

This info belongs in a wiki until a way is found how to request that permission from MIUI 11, should it be possible.

**Comment by PhyberApex** [comment]
Just as an update I am still not able to get the prompt from OpenKeychain. I have tried many settings by now but I still can't get it to show. The settings shown in the screenshot is enabled on my device.

~Cheers

**Comment by justinacolmena** [comment]
The Openkeychain app is highly targeted by vicious gangs of crooked cops, Bitcoin thieves and politicians who scream "Darknet!" whenever they see an encryption app. 

https://www.reddit.com/r/darknet/comments/f2nrw7/openkeychain/

It's too political. The cops who dictate the design and software of these devices refuse to allow a middle ground between all manner of wholesale identity theft and drug dealing, in which they are actively complicit, and their universal warrant for plant-and-bust child porn or DMCA or whatever else they can use to press arbitrary and unfounded criminal charges.

On February 11, 2020 10:13:35 PM AKST, PhyberApex <notifications@github.com> wrote:
>Just as an update I am still not able to get the prompt from
>OpenKeychain. I have tried many settings by now but I still can't get
>it to show. The settings shown in the screenshot is enabled on my
>device.
>
>~Cheers
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/k9mail/k-9/issues/2601#issuecomment-585064112

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by PhyberApex** [comment]
> The Openkeychain app is highly targeted by vicious gangs of crooked cops, Bitcoin thieves and politicians who scream "Darknet!" whenever they see an encryption app. https://www.reddit.com/r/darknet/comments/f2nrw7/openkeychain/ It's too political. The cops who dictate the design and software of these devices refuse to allow a middle ground between all manner of wholesale identity theft and drug dealing, in which they are actively complicit, and their universal warrant for plant-and-bust child porn or DMCA or whatever else they can use to press arbitrary and unfounded criminal charges.
> […](#)
> On February 11, 2020 10:13:35 PM AKST, PhyberApex ***@***.***> wrote: Just as an update I am still not able to get the prompt from OpenKeychain. I have tried many settings by now but I still can't get it to show. The settings shown in the screenshot is enabled on my device. ~Cheers -- You are receiving this because you are subscribed to this thread. Reply to this email directly or view it on GitHub: [#2601 (comment)](https://github.com/k9mail/k-9/issues/2601#issuecomment-585064112)
> -- Sent from my Android device with K-9 Mail. Please excuse my brevity.

Thank you for your very helpful and absolutely relevant reply .__."

**Comment by myasn1k** [comment]
In Xiaomi phones (MIUI) you have to do this more or less: https://forum.k9mail.app/t/how-do-i-make-k-9-mail-work-with-openkeychain-on-xiaomis-miui/82

Now it works!

**Comment by Anarbb** [comment]
Did everything and still doesn't work, Maybe it doesn't on chinese phones, just drop OKC and implement PGP into the app like what thunderbird did please, or have it as an option.

**Comment by Anarbb** [comment]
Did everything and still doesn't work, Maybe it doesn't on chinese phones, just drop OKC and implement PGP into the app like what thunderbird did please, or have it as an option.
This has been a 5 year bug and still hasn't been fixed holy.

**Comment by jsinfate** [comment]
I can confirm that this is still an issue today. I am unable to add my keys through OpenKeychain in v6.603 or pre-release v6.714. I have done all battery optimizations for both programs allowing them all access. Uninstalled and reinstalled each version and tested and nothing happens when selecting my PGP key. Several times I received an error that OpenKeychain stopped running. 
Is there a way we can make this a higher priority?

**Comment by TW-Bit** [comment]
Similar issue, using:
Version 6.0.4 
installed from F-droid
Google Pixel 8a
GrapheneOS
System language is English

I did a reinstall, didn't work. When I try to confirm with fingerprint, I can't select a key in the menu. Don't know if the previous commenter's suggestions are useful to me as I see no menu/permission option for that in Graphene's settings for apps. Shame that the bug has gon on this long...
